File: features/lender/src/main/java/com/nivasafinance/features/lender/lenderoffice/service/LenderOfficeReadService.java
package com.nivasafinance.features.lender.lenderoffice.service;

import com.nivasafinance.common.base.model.PaginatedResponse;
import com.nivasafinance.common.base.model.PaginationRequest;
import com.nivasafinance.features.lender.lenderoffice.dto.LenderOfficeReponseData;
import com.nivasafinance.features.lender.lenderoffice.dto.LenderOfficeSearchRequest;
import com.nivasafinance.features.lender.lenderoffice.enums.LenderOfficeStatus;

import java.util.List;
import java.util.UUID;

public interface LenderOfficeReadService {
    LenderOfficeReponseData getByKey(String key);
    LenderOfficeReponseData getById(UUID id);
    List<LenderOfficeReponseData> getByLenderKeyAndStatus(String lenderKey, LenderOfficeStatus status);
    PaginatedResponse<LenderOfficeReponseData> getLenderOfficesPaginated(String lenderKey, PaginationRequest paginationRequest,
                                                                         LenderOfficeStatus status);
    PaginatedResponse<LenderOfficeReponseData> searchLenderOffices(String lenderKey, PaginationRequest paginationRequest,
                                                                   LenderOfficeSearchRequest request);
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lenderoffice/service/LenderOfficeReadService.kt
package com.nivasafinance.features.lender.lenderoffice.service

import com.nivasafinance.features.lender.lenderoffice.dto.LenderOfficeReponseData
import com.nivasafinance.features.lender.lenderoffice.enum.LenderOfficeStatus
import java.util.UUID

interface LenderOfficeReadService {
    fun getByKey(key: String): LenderOfficeReponseData
    fun getById(id: UUID): LenderOfficeReponseData
    fun getByLenderKeyAndStatus(lenderKey: String, status: LenderOfficeStatus): List<LenderOfficeReponseData>
}
